converter/dicom-cast/src/Microsoft.Health.DicomCast.TableStorage/Features/Storage/TableSyncStateStore.cs

```diff
@@ -0,0 +1,70 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Cosmos.Table;
+using Microsoft.Health.DicomCast.Core.Exceptions;
+using Microsoft.Health.DicomCast.Core.Features.State;
+using Microsoft.Health.DicomCast.TableStorage.Features.Storage.Models.Entities;
+
+namespace Microsoft.Health.DicomCast.TableStorage.Features.Storage.Models
+{
+    public class TableSyncStateStore : ISyncStateStore
+    {
+        private readonly CloudTableClient _client;
+
+        public TableSyncStateStore(CloudTableClient client)
+        {
+            EnsureArg.IsNotNull(client, nameof(client));
+
+            _client = client;
+        }
+
+        public async Task<SyncState> ReadAsync(CancellationToken cancellationToken = default)
+        {
+            CloudTable table = _client.GetTableReference(Constants.SyncStateTableName);
+            TableOperation retrieveOperation = TableOperation.Retrieve<SyncStateEntity>(Constants.SyncStatePartitionKey, Constants.SyncStateRowKey);
+
+            TableResult result;
+
+            try
+            {
+                result = await table.ExecuteAsync(retrieveOperation);
```

Comment:
`ExecuteAsync` appears to have an overload that accepts `CancellationToken`, so we should also use that to propogate the cancellation if desired. If we do that, it also means re-throwing any `OperationCanceledException`

Reply:
Thanks for the catch